tracing-opentelemetry/src/lib.rs

```diff
@@ -30,6 +30,10 @@
 //!     let root = span!(tracing::Level::TRACE, "app_start", work_units = 2);
 //!     let _enter = root.enter();
 //!
+//!     // Optionally set a dynamic span name for otel to export
+//!     let operation = "GET http://example.com".to_string();
+//!     span!(tracing::Level::TRACE, "http_request", otel.name=operation.as_str());
```

Comment:
nit: i think the `to_string`/`as_str` can be removed?
```suggestion
//!     let operation = "GET http://example.com";
//!     span!(tracing::Level::TRACE, "http_request", otel.name = operation);
```

Reply:
True, was trying to signal that it worked for arbitrary strings, not merely 'static str... maybe there is a better way of conveying that?

Reply:
Hmm, that makes sense. I'm fine with leaving it as-is, then. Or, we could add multiple examples with string constants, string slices, etc?

Reply:
As this is the first example that people will see when they look at the docs it's probably best to keep it simple, could remove it from the example maybe if it is confusing? less necessary to show usage now that there is a list of the special fields.

Reply:
Removed for first time user experience simplicity as there is now both docs and an example test if folks look for how to do this.